Flaky Panic in unit test

**Describe the bug**
`TestNew` in pkg/reconciler/broker/controller_test.go is flaky.

https://github.com/google/knative-gcp/blob/2355b9355828bd1e93c0b154952f91ae22a87f69/pkg/reconciler/broker/controller_test.go#L44

It occassionally fails with:

```shell
panic: Log in goroutine after TestNew has completed

goroutine 61 [running]:
testing.(*common).logDepth(0xc00012c000, 0xc0006a60b0, 0xa3, 0x3)
        /usr/local/go/src/testing/testing.go:680 +0x4d3
testing.(*common).log(...)
        /usr/local/go/src/testing/testing.go:662
testing.(*common).Logf(0xc00012c000, 0x281cbc9, 0x2, 0xc0006a0040, 0x1, 0x1)
        /usr/local/go/src/testing/testing.go:701 +0x7e
go.uber.org/zap/zaptest.testingWriter.Write(0x2c0f440, 0xc00012c000, 0x0, 0xc000ba4400, 0xa4, 0x400, 0xc000ba4400, 0x24eaa20, 0xc0006a0030)
        /Users/harwayne/go/src/github.com/google/knative-gcp/vendor/go.uber.org/zap/zaptest/logger.go:130 +0xfc
go.uber.org/zap/zapcore.(*ioCore).Write(0xc0009ffad0, 0xff, 0xbfadd3616a5e63c8, 0x2322442, 0x3b45a80, 0x282e22d, 0x11, 0xc000610120, 0x1d, 0x1, ...)
        /Users/harwayne/go/src/github.com/google/knative-gcp/vendor/go.uber.org/zap/zapcore/core.go:90 +0x106
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc000584a50, 0x0, 0x0, 0x0)
        /Users/harwayne/go/src/github.com/google/knative-gcp/vendor/go.uber.org/zap/zapcore/entry.go:216 +0x117
go.uber.org/zap.(*SugaredLogger).log(0xc000010090, 0xc0001a17ff, 0x0, 0x0, 0xc0001a16f0, 0x1, 0x1, 0x0, 0x0, 0x0)
        /Users/harwayne/go/src/github.com/google/knative-gcp/vendor/go.uber.org/zap/sugar.go:234 +0x100
go.uber.org/zap.(*SugaredLogger).Debug(...)
        /Users/harwayne/go/src/github.com/google/knative-gcp/vendor/go.uber.org/zap/sugar.go:97
github.com/google/knative-gcp/pkg/reconciler/broker.(*Reconciler).TargetsConfigUpdater(0xc000a08140, 0x2bff220, 0xc0009ff6e0)
        /Users/harwayne/go/src/github.com/google/knative-gcp/pkg/reconciler/broker/broker.go:492 +0xa3
created by github.com/google/knative-gcp/pkg/reconciler/broker.NewController
        /Users/harwayne/go/src/github.com/google/knative-gcp/pkg/reconciler/broker/controller.go:102 +0x770
FAIL    github.com/google/knative-gcp/pkg/reconciler/broker     0.068s
FAIL
```

This appears to be caused by calling the logger on this line, after the test has finished.

https://github.com/google/knative-gcp/blob/2355b9355828bd1e93c0b154952f91ae22a87f69/pkg/reconciler/broker/broker.go#L494

While the test is not currently closing the context, and it probably should, I don't think that will sufficient to prevent the panic because that function seems to be running after the test finishes and doesn't check for context closure until after logging (and even after noticing the context is closed, logs again).

 **Expected behavior**
Unit test passes consistently.

 **To Reproduce**
```shell
go test ./pkg/reconciler/broker -count 100 -run 'TestNew$'
```

 **Additional context**
Prow failure: https://prow.knative.dev/view/gcs/knative-prow/pr-logs/pull/google_knative-gcp/1183/pull-google-knative-gcp-unit-tests/1267937432632823810

--- a/pkg/reconciler/broker/broker.go
+++ b/pkg/reconciler/broker/broker.go
@@ -491,7 +491,6 @@ func (r *Reconciler) LoadTargetsConfig(ctx context.Context) error {
 }
 
 func (r *Reconciler) TargetsConfigUpdater(ctx context.Context) {
-	r.Logger.Debug("Starting TargetsConfigUpdater")
 	// check every 10 seconds even if no reconciles have occurred
 	ticker := time.NewTicker(targetsCMResyncPeriod)
 
@@ -501,7 +500,6 @@ func (r *Reconciler) TargetsConfigUpdater(ctx context.Context) {
 	for {
 		select {
 		case <-ctx.Done():
-			r.Logger.Debug("Stopping TargetsConfigUpdater")
 			return
 		case <-r.targetsNeedsUpdate:
 			if err := r.updateTargetsConfig(ctx); err != nil {

